addons/mass_mailing/static/src/js/mass_mailing_widget.js

```diff
@@ -65,6 +66,10 @@ var MassMailingFieldHtml = FieldHtml.extend({
             self._isDirty = self.wysiwyg.isDirty();
             await self._doAction();
 
+            if (isCodeview) {
+                $editable.html(self.value);
+            }
+
```

Comment:
I would place this code right after 
```js
var isCodeview = this._$codeview && !this._$codeview.hasClass('d-none');
```

So that it gets cleaned, and it calls `saveModifiedImages`.
Otherwise, if edit the dom, add some image through the UI, switch to codeview, then save, it would not be cleaned and the images would not be saved properly.

Reply:
done, thanks